datadog-integrations-aws-handler/src/datadog_integrations_aws/handlers.py

```diff
@@ -227,6 +249,8 @@ def read_handler(
     model.HostTags = aws_account.host_tags
     model.FilterTags = aws_account.filter_tags
     model.AccountSpecificNamespaceRules = aws_account.account_specific_namespace_rules
+    if model.ExternalIDSecretName is None:
+        model.secret_name = DEFAULT_SECRET_NAME
```

Comment:
I don't think this would work. In the read handler, you would never have access to the configured `ExternalIDSecretName`, it would always be None. So it's not possible to have a user specify a different name.

Reply:
Otherwise we'd have stored the external ID here directly from the start when it's returned to us by the API.
There's unfortunately no sort of "state" in cloudformation

Reply:
This is set by the user - we aren't using this value as stored state. It's just like how we read in the AccountId and such from the model above. It works, I tested it.

Reply:
Does it work when you pass in a custom name in the template, i.e. you can reference it in an other resource for instance and you receive the custom name ? 
Because AFAIK, the only thing that is passed to the read handler by cloud formation is the primary identifier of the resource, and everything else is `None` basically. So you'd always return the default name here.

Reply:
Ah, I think I only checked that it created the resource with the right name, not that it read it with the right name. I don't think we even need to return the secret_name in this case as the user knows what name they've given it if they've passed it in. I'll remove this.

Reply:
Yeah ok, makes sense if you're expecting users to hardcode the given secret name to use it somewhere else.
Let's properly document what the default name is then, and that it won't be possible to reference that attribute.